### ISSUE TITLE
Add a interface to report current progress

### ISSUE THREAD

**Issue by zhou13** [jborg/attic#115]
It will help a lot if attic can report more information such as ETA while creating its backup.  Currently, `-v` will make too many noise and withother `-v` it is too silent.

I hope that attic can have a progress bar, just like downloading using wget, to report the number of files back up / total number of files.

**Comment by leoetlino** [comment]
I used the following command to have a nice progress bar during a backup. It is not 100% precise and it can certainly be improved, but it does the job.

```
stdbuf -oL -eL attic create --verbose \
    $REPOSITORY::(date +%Y-%m-%d%z) \
    $WHAT_TO_BACKUP  | pv -fpterl -s $NUMBER_OF_FILES_TO_PROCESS > /dev/null
```

Attic prints a line per processed file in verbose mode, so we can use `pv` for the progress bar.

Of course, the initial step of counting the files will make the backup take some more time. It takes a few minutes on my PC, so depending on your case, this solution might not be acceptable.

```
0:02:12 [35,9 /s] [=>                                                  ]  5% ETA 0:35:15
^ total time  ^ files per second (average)
```

**Comment by anarcat** [comment]
So yeah i was thinking of something like this, but i was wondering if
attic had inside knowledge of the file sizes, for example, which would
be a better estimate of ETAs than the number of files.

Having that kind of output without resorting to complex pipelines would
also be a must here, in my opinion. having that by default would greatly
improve adoption as it gives a great "wow" effect to newcomers.

it is also a serious plus for sysadmins wishing to do a quick
snapshot...

a.

**Comment by jmiserez** [comment]
Printing or updating progress for every file can be a major slowdown, I have noticed this specifically with `rsync`: When run with `--progress` it slows to a crawl compared to the regular `-v` (which lists the files same as attic's `-v`), especially when lots of small files are involved. At the very least progress updates should be limited to once every few seconds.

**Comment by anarcat** [comment]
yeah, obviously i don't want to have one line per file. i am thinking more of a progress-bar style thing, something like pv or wget would show.

**Comment by anarcat** [comment]
here's an example of a backup done with bup:

```
Saving: 12.33% (42605417/345558958k, 38711/2413360 files) 2h33m 32000k/s
```

This line updates a few times per second, probably using `\r` to avoid flooding the terminal. it also turns itself off if the output is not a terminal (so will not pollute cronjob outputs either). quite useful.

**Comment by jmiserez** [comment]
@anarcat From looking at the attic source it seems like attic has no way of knowing the number of files when it starts a backup for the first time, and counting them all beforehand seems like unnecessary overhead.

@leoetlino You can get the number of files in an archive by running `attic info` for a given archive, that is assuming you already have an initial backup.

To get a bit nicer progress info, you can use named pipes instead of `/dev/null` to get both the list of files and the progress bar at the same time, in two terminals. And finally you can make `pv` wait for the first line by adding the `-W` parameter.

So on one terminal:
`mkfifo /tmp/foo-pipe`
`cat /tmp/foo-pipe`

And on the other terminal:
`stdbuf -oL -eL attic create -v $NAME $PATHS  | pv -fpterlW -s $NUMFILES > /tmp/foo-pipe`

**Comment by anarcat** [comment]
@jmiserez i would assume that the "cache" it builds up when it starts has that information. but i haven't read all the source.

**Comment by ThomasWaldmann** [comment]
see PR #250.
